wsmaster/che-core-api-factory-bitbucket/src/main/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketUrl.java

```diff
@@ -147,6 +148,14 @@ public String getHostName() {
     return "https://" + HOSTNAME;
   }
 
+  @Override
+  public Optional<String> getCredentials() {
+    if (!isNullOrEmpty(username) && super.getUrl().getUserInfo().equals(username)) {
```

Comment:
What if `username` is a PAT ?
How do we handle this case?

Reply:
PAT is not supposed to be here. We don't support repository URL with PAT but we do support devfile url with PAT. The devfile url e.g. `https://bitbucket.org/ivinokur/public/raw/HEAD/devfile.yaml` will not be handled by the `BitbucketUrl` because it doesn't meet the regex pattern rules. It is handled by the `DefaultFactoryUrl` which doesn't have this check.